storage/imx6ull-flash/flashsrv.c

```diff
@@ -114,6 +124,82 @@ static int flashsrv_devfsSymlink(const char *name, const char *target)
 }
 
 
+/* Partition table access functions (ptable buffer must have mtd->writesz size) */
+
+static int flashsrv_ptableRead(storage_t *strg, ptable_t *ptable)
+{
+	storage_mtd_t *mtd = strg->dev->mtd;
+	size_t retlen;
+	off_t offs;
+	int err, i;
+
+	/* Find first valid partition table copy */
+	for (i = 0; i < PTABLE_NCOPIES; i++) {
+		offs = strg->size - (i + 1) * mtd->erasesz;
+		if (mtd->ops->block_isBad(strg, offs)) {
+			continue;
+		}
+
+		err = mtd->ops->read(strg, offs, ptable, mtd->writesz, &retlen);
+		/* -EUCLEAN isn't a fatal error (indicates dangerous page degradation but all bitflips were successfully corrected) */
+		if (((err < 0) && (err != -EUCLEAN)) || (retlen != mtd->writesz)) {
+			continue;
+		}
+
+		/* Partition table must fit in a NAND page */
+		if (ptable_size(ptable->count) > mtd->writesz) {
```

Comment:
Disputable nitpick - it might be worth inserting le32toh for count here, so there won't be any "got ya" if someone copy this solution to their code

Reply:
Done.